src/gui/DatabaseWidget.cpp

```diff
@@ -370,7 +370,9 @@ void DatabaseWidget::replaceDatabase(QSharedPointer<Database> db)
     // TODO: instead of increasing the ref count temporarily, there should be a clean
     // break from the old database. Without this crashes occur due to the change
     // signals triggering dangling pointers.
+#if defined(WITH_XC_KEESHARE)
     auto oldDb = m_db;
+#endif
```

Comment:
Don't guard this, it is required to maintain the database pointer during the life of this function. Using it with KeeShare is an added benefit (see the TODO above it).